src/interfaces/ParamsWithId.ts

```diff
@@ -0,0 +1,13 @@
+import { z } from 'zod';
+import { ObjectId } from 'mongodb';
+
```

Comment:
**nitpick:** You split your code very nicely, but I suggest to keep all domain specific interfaces in their directories. I mean, create `card.interfaces.ts`, and move there all Card related interfaces from this general directory